src/v/cluster/partition.cc

```diff
@@ -559,6 +579,18 @@ ss::future<> partition::update_configuration(topic_properties properties) {
             _archiver->notify_topic_config();
             co_await _archiver->start();
         }
+
+        if (_cloud_storage_partition) {
+            co_await _cloud_storage_partition->stop();
```

Comment:
This is an expensive operation. Do we really need to do this upon every topic configuration change? Some topic configuration changes are not related to cloud storage at all.

Reply:
Also, maybe it's reasonable to start a bg operation here instead of waiting until remote_partition is stopped.